brave/brave-ios
# Fix #5723: Solana transaction support in Transaction Parser

## Summary of Changes
- Support the 3 Solana transaction types in the `TransctionParser`
- Fix `AccountActivityStore` using hardcoded `.eth` coin type, causing issues when connected to other networks / viewing Solana accounts
- Fix `AssetDetailStore` using hardcoded `.eth` coin type, causing issues displaying transactions

This pull request fixes #5723

## Submitter Checklist:


## Test Plan:
1. Create a send transaction for SOL token
2. Create a send transaction for any non-SOL token on Solana (Spl Token)
3. Verify transactions show up in account detail / account activity for the respective account that made the transaction
4. Verify transactions show up in the asset detail for the respective tokens (SOL and the non-SOL token)

You can use [Solana explorer](https://explorer.solana.com/) to verify transaction display.

## Screenshots:

![#5723 - account details](https://user-images.githubusercontent.com/5314553/180493319-114c6776-4e13-47bc-ab20-371dc7e0e76e.png) | ![#5723 - asset details SOL](https://user-images.githubusercontent.com/5314553/180493533-1e109e17-91c5-46fb-b572-9521eaceab62.png) | ![#5723 - asset details STEPHEN](https://user-images.githubusercontent.com/5314553/180493339-58b68ac5-f854-4c7c-8bba-43b247acc6be.png)
--|--|--

[SOL send transaction on explorer](https://explorer.solana.com/tx/5BV3TMCuAKgw5NfGDdQMvThfETsN62fpJhhSkGHXA3FnBGyXp98A9D3EDZPZjfEo3LqTJzqnEujJ6rQdAGSgkyAL?cluster=devnet)
[STEPHEN send transaction on explorer](https://explorer.solana.com/tx/4S8er1q5PsbKyhQrXFt1E7Y1VUJZVKDVVvQatXYrnz96ZuZ9T2QZsdV7fsudgxmYK7FGD2AiyQtwmRTDGHnf4miU?cluster=devnet)
(transactions from screenshots)


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
